src/Crypto/Signing/HsmSignerService.cs

```
@@ -0,0 +1,178 @@
+            IHsmSignerConfig config
+            )
+        {
+            _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+            _config = config ?? throw new ArgumentNullException(nameof(config));
+
+            _httpClient.BaseAddress = new Uri(_config.BaseAddress);
+            _httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+        }
+
+        public async Task<byte[]> GetNlSignatureAsync(byte[] content)
+        {
+            using var sha256Hasher = SHA256.Create();
+            var contentHash = sha256Hasher.ComputeHash(content);
+            var contentHashBased64 = Convert.ToBase64String(contentHash);
+
+            var certificateChain = GetCertificateChain();
+            var nlPem = _config.NlPem;
+
+            var builder = new StringBuilder();
+
+            // Add EV RSA certificate
+            builder.AppendLine(nlPem);
+
+            // Add EV certificate chain
+            foreach (var cert in certificateChain)
+            {
+                builder.AppendLine(new string(PemEncoding.Write("CERTIFICATE", cert.RawData)));
+            }
+
+            var pem = builder.ToString();
+
+            var pemBytes = Encoding.UTF8.GetBytes(pem);
+            var pemBytesBased64 = Convert.ToBase64String(pemBytes);
+
+            var requestModel = new HsmSignerRequestModel()
+            {
+                Algorithm = "sha256",
+                Cert = pemBytesBased64,
+                Hash = contentHashBased64,
+                TimeStamp = true
+            };
+
+            _httpClient.DefaultRequestHeaders.Remove("Authorization");
+            _httpClient.DefaultRequestHeaders.Add("Authorization", _config.NlJwt);
+
+            var response = _httpClient.PostAsJsonAsync("/cms", requestModel).Result;
+
+            if (!response.IsSuccessStatusCode)
+            {
+                //TODO: error handling
+                return Array.Empty<byte>();
```

Could we just throw here? To prevent invalid signatures?

---

Yeah, good point. Throwing some semi-random exceptions for now, so we can get testing 😅